woocommerce/woocommerce-ios
# Bulk Editing: Add price picker

Closes: https://github.com/woocommerce/woocommerce-ios/issues/8520

## Description

This PR adds status selector for bulk editing on Products List. Networking action is connected, loading, success and error states implemented.

### How

UI is based on `BulkUpdatePriceViewController` that is used for bulk variations update. `PriceInputViewController` is mostly the same, but stripped of networking actions and variation logic. In future we can unify them by adding some coordinator/subclass for variations.

## Testing

1. Build and run the app in debug/alpha mode.
2. On the products list tap the "multi-select" icon in the navbar.
3. Select a few products from the list.
4. Tap "Bulk update" in the bottom toolbar.
5. Tap "Update price".
6. Confirm that correct price is described in footer only if all selected products have the same status.
7. Confirm that "Apply" button is disabled for empty price.
8. Confirm that "Apply" button is enabled when price is entered.
9. Enter a price that is less than sale price of one of the selected products.
10. Tap "Apply". Observe immediate validation error notice.
11. Enter a valid price (greater than any of sale prices).
12. Tap "Apply". Observe loading state and success message.
13. Confirm products are updated in the list.


## Video

https://user-images.githubusercontent.com/3132438/213264489-fb8447f8-f55d-43c1-9c8d-923be44129bb.mp4

---
